cartographer/io/points_batch.cc

```diff
@@ -19,17 +19,29 @@
 namespace cartographer {
 namespace io {
 
-void RemovePoints(std::vector<int> to_remove, PointsBatch* batch) {
-  std::sort(to_remove.begin(), to_remove.end(), std::greater<int>());
-  for (const int index : to_remove) {
-    batch->points.erase(batch->points.begin() + index);
+void RemovePoints(std::unordered_set<int> to_remove, PointsBatch* batch) {
+  std::vector<Eigen::Vector3f> points;
+  points.reserve(batch->points.size());
+  std::vector<float> intensities;
+  intensities.reserve(batch->intensities.size());
```

Comment:
here is guess you have to check and only reserve batch->intensities.size() - to_remove.size()) if batch->intensities.size() > 0